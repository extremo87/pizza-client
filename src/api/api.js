import axios from 'axios';


const ServerResponseCode = {
  UNAUTHORIZED: 401,
  VALIDATION: 422
};

export const createAPI = (onError) => {
  const api = axios.create({
    baseURL: process.env.API_URL,
    timeout: 5000
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (error) => {
    if (error.response) {
      if (error.response.status === ServerResponseCode.VALIDATION) {
        const errors = [];

        Object.keys(error.response.data.errors).map((key) => {
          errors.push(error.response.data.errors[key]);
        });

        onError(errors.join(`,`));

      } else {
        onError(error.message);
      }
    } else {
      onError(error.message);
    }
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
