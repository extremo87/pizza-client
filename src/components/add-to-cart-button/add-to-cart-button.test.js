import React from 'react';
import AddToCartButton from './add-to-cart-button';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import NameSpace from "../../reducers/name-space.js";
import thunk from 'redux-thunk';
import {createAPI} from "../../api/api";


const api = createAPI(() => {});

const mockStore = configureStore([thunk.withExtraArgument(api)]);

const product = {
  id: 1,
  title: `MARGHERITA`,
  ingredients: `Tomatoes, mozzarella, ricotta, bacon, onion`,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
  image: `https://static.nnc.kz/pizza1.jpg`,
  price: 18,
  created_at: `2020-09-21 13:18:35`,
  updated_at: `2020-09-21 13:18:35`
};


const store = mockStore({
  [NameSpace.CART]: {
    cart: {},
  }
});


it(`Add to cart button conponent renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <AddToCartButton
            addToCart={() => {}}
            increment={() => {}}
            decrement={() => {}}
            product={product}
            cart={{}}
          />
        </Provider>

    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
