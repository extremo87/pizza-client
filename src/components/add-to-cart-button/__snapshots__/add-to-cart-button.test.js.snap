// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add to cart button conponent renders correctly 1`] = `
<div
  className="btn-group"
>
  <button
    className="btn btn-sm btn-outline-secondary add-to-cart"
    onClick={[Function]}
    type="button"
  >
    Add to cart
  </button>
</div>
`;
